Add: Article requirements

prairir opened the pull request:
### Did you read and follow article requirements

This is literally for writing them

### Did you run prettier

Yes

### What are you trying to accomplish?

Add article requirements this includes citing sources and dating sections/articles

### How are you accomplishing it?

Uhh theres a lot to it

### Is there anything reviewers should know?

LMK what you think, this is still very early stages

-   [x] Is it safe to rollback this change if anything goes wrong?

prairir committed:
Add: Article requirements

I added article requirements. These are requirements on what to do in an
article. This currently includes citing sources and dating last update times

kilbouri commented on REQUIREMENTS.md:
```diff
@@ -0,0 +1,128 @@
+
+## Citations
+
+As most of this information is factural, citating of sources is needed.
```
factual*
citing*

kilbouri commented on REQUIREMENTS.md:
```diff
@@ -0,0 +1,128 @@
+
+### Sections
+
+**Use Case:** If you need to cite a sections
```
section*

kilbouri reviewed (changes requested):
A few typos to fix. See comments on commit. Otherwise, looks great.

jere-mie commented on REQUIREMENTS.md:
````diff
@@ -0,0 +1,128 @@
+_Last updated on 2020/09/14_
+```
+
+### Articles With Sections
````
At what point do you not put a date on a sub section? You have it for h1 and h2, but didn't specify at what point you stop putting last updated dates.

Frankly, I think it's a bit overkill to put "last updated" dates on subsections (h2), and it may even be overkill for h1s as well. At the very least, if we need to have last updated dates for sections, I say we stop at h1s.

prairir replied:
Im fine with leaving it at h1 until there is a need to go more

prairir committed:
Update: Clarify date modified scope and fix typos

Data modified was originally scopped to all sections, that can be a huge
headache. To fix this, Jeremie suggested that we stop at h1. I think h1
is perfect unless we get into weirdness with article formatting. Then i
think we should include h2. Either way, its fine for now.

nysdas commented on REQUIREMENTS.md:
```diff
@@ -0,0 +1,128 @@
+
+As most of this information is time sensitive, listing the last updated date is important.
+
+Dates need to be in the format `yyyy/mm/dd` for example, `2022/12/25`
```
How we will maintain the last updated info to be **updated**? I would recommend to add in the list of check marks when someone send a PR. That way, both committer and reviewer will be responsible for the details. 

**If you updated an Article, did you update the last updated section?**

kilbouri replied:
I would think that is implied by the "did you read the article requirements" question

nysdas replied:
adding another check mark will make sure, it has been done. It was just something I thought.

prairir replied:
yeah it is implied by the `did you read the article requirements` question. There may end up being a bunch of article requirements with their own rules and conventions. If we put them all as check boxes in the PR template, it can get complicated. 

I think having a separate document is the way to go.

nysdas replied:
to be honest, I don't think three or four check boxes will hurt.

- [ ] Read the Article Requirements 
- [ ] prettier 
- [ ] last updated

prairir replied:
Yeah but `last updated` is included in `Read the Article Requirements`. I just think its redundant. Not to mention we can change it if it stops working out for us. I do like the `prettier` idea though

kilbouri replied:
The "prettier" box is solid IMO, but I still think Last Updated is implied by Article Requirements. Its not an issue with the number of boxes we would have now, its that if we start listing the article requirements in the PR template, then eventually we'll have dozens of them.

nysdas replied:
agree!!
